### ISSUE TITLE
Ctrl+C and Ctrl+X no longer work in forms for Notepad++ 8.6.1 and newer.

### ISSUE THREAD

**Issue by molsonkiko** [young-developer/nppNavigateTo#64]
__To replicate the issue:__

1. Install [NavigateTo 2.7.0](https://github.com/young-developer/nppNavigateTo/releases/tag/v.2.7.0) on [Notepad++ 8.6.1](https://notepad-plus-plus.org/downloads/v8.6.1/) (32bit or 64bit, shouldn't matter).
2. Copy some text to the clipboard (call this `Original selection`)
3. Open the NavigateTo window in any document.
4. Select some text in the NavigateTo textbox (call this `NavigateTo selection`)
5. Copy the NavigateTo textbox selection using Ctrl+C.
6. Observe that `Original selection` is still in the clipboard.
7. Cut the NavigateTo textbox selection using Ctrl+X.
8. Observe that `Original selection` is still in the clipboard.
9. *If desired*, try doing similar things on the settings form. You should find that Ctrl+C and Ctrl+X are broken everywhere.

__My proposed solution__

You can fix this issue by following the process I describe in my [NppCSharpPluginPack](https://github.com/molsonkiko/NppCSharpPluginPack/blob/main/docs/README.md#registering-and-unregistering-forms-with-nppm_modelessdialog).

If you like, I can submit a PR. While this issue *isn't hard* to fix, it *is annoying and finicky* to fix because you have to visually compare the order of `Controls.Add` calls to the desired tab order for every form in your plugin.

**Comment by young-developer** [comment]
Hm. I tested it on my machine with v.8.6.2 and I am able to copy and cut text from searchbox without any issues.

```
Notepad++ v8.6.2   (64-bit)
Build time : Jan 14 2024 - 02:16:00
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : ON
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 11 Home (64-bit)
OS Version : 23H2
OS Build : 22631.3085
Current ANSI codepage : 1252
Plugins : 
    mimeTools (3)
    NavigateTo (2.7)
    NppConverter (4.5)
    NppExport (0.4)

```

**Comment by young-developer** [comment]
![image](https://github.com/young-developer/nppNavigateTo/assets/4460168/d084827b-1ae4-4f18-998d-35cb57a27147)

**Comment by molsonkiko** [comment]
@young-developer 
> Hm. I tested it on my machine with v.8.6.2 and I am able to copy and cut text from searchbox without any issues.

Yes, I'm aware of this. However, if you go to `Settings->Preferences...->Editing` and uncheck `Enable Copy/Cut line without selection` you will be able to reproduce this bug, unless you have text selected in your current document.

**Comment by young-developer** [comment]
I double-checked in 8.6.5 and I am still able to copy without any issues. @molsonkiko please could you retest?

**Comment by molsonkiko** [comment]
The issue only manifests when you do the following:

1. Select `Setting->Preferences` from the main menu
2. Go to `Editing 1` tab
3. Uncheck `Enable Copy/Cut Line without selection`
4. Make sure you do not have any text selected
5. Try to use Ctrl+X or Ctrl+C to copy or cut from FrmNavigateTo or from the settings form

I do not know that it's really worthwhile to fix this issue, given what a small corner case this is. [As noted here](https://github.com/molsonkiko/NppCSharpPluginPack/blob/main/docs/README.md#registering-and-unregistering-forms-with-nppm_modelessdialog) there are a bunch of disadvantages associated with fixing it.

**Comment by webbrain-one** [comment]
Hi there! 👋 I've opened PR #71 to fix the `Ctrl+C` and `Ctrl+X` shortcut issue in the plugin's forms. This update resolves the regression in Notepad++ 8.6.1+ by properly registering the forms with `NppMsg.Nppm_modelessDialog`, which restores standard clipboard functionality across all forms. Please take a look when you have a moment, and let me know if you have any questions or need adjustments! 😊
